src/fides/api/task/create_request_tasks.py

```diff
@@ -429,41 +429,59 @@ def run_access_request(
-                session, privacy_request, traversal_nodes, erasure_end_nodes, graph
+        try:
+            logger.info("Building access graph for {}", privacy_request.id)
+            traversal: Traversal = Traversal(graph, identity)
+
+            # Traversal.traverse populates traversal_nodes in place, adding parents and children to each traversal_node.
+            traversal_nodes: Dict[CollectionAddress, TraversalNode] = {}
+            end_nodes: List[CollectionAddress] = traversal.traverse(
+                traversal_nodes, collect_tasks_fn
+            )
+            # Save Access Request Tasks to the database
+            ready_tasks = persist_new_access_request_tasks(
+                session, privacy_request, traversal, traversal_nodes, end_nodes, graph
             )
 
-        # cache a map of collections -> data uses for the output package of access requests
-        privacy_request.cache_data_use_map(
-            format_data_use_map_for_caching(
-                {
-                    coll_address: tn.node.dataset.connection_key
-                    for (coll_address, tn) in traversal_nodes.items()
-                },
-                connection_configs,
+            if (
+                policy.get_rules_for_action(action_type=ActionType.erasure)
+                and not privacy_request.erasure_tasks.count()
+            ):
+                # If applicable, go ahead and save Erasure Request Tasks to the Database.
+                # These erasure tasks aren't ready to run until the access graph is completed
+                # in full, but this makes sure the nodes in the graphs match.
+                erasure_end_nodes: List[CollectionAddress] = list(graph.nodes.keys())
+                persist_initial_erasure_request_tasks(
+                    session, privacy_request, traversal_nodes, erasure_end_nodes, graph
+                )
+
+            # cache a map of collections -> data uses for the output package of access requests
+            privacy_request.cache_data_use_map(
+                format_data_use_map_for_caching(
+                    {
+                        coll_address: tn.node.dataset.connection_key
+                        for (coll_address, tn) in traversal_nodes.items()
+                    },
+                    connection_configs,
+                )
             )
-        )
+        except TraversalError as err:
+            logger.error(
+                "TraversalError encountered for privacy request {}. Error: {}",
+                privacy_request.id,
+                err,
+            )
+            errors = ", ".join(err.errors)
+            privacy_request.add_error_execution_log(
+                session,
+                connection_key=errors,
+                dataset_name=errors,
+                collection_name=errors,
+                message=f"{err}",
+                action_type=ActionType.access,
+            )
+            privacy_request.error_processing(session)
+            raise err
```

Comment:
Did you test this for DSR 3.0 as well? Your instructions are for DSR 2.0 -

This involves setting the following env vars:
```
[execution]
use_dsr_3_0 = true
```

```
[celery]
task_always_eager = false
```

and then running:
```
nox -s dev -- shell postgres worker
```

I keep running into a separate issue on the worker container, I want to make sure this isn't just me:

```
./../data/language/languages.yml from /fides/.fides/fides.toml
> Loaded config from: /fides/.fides/fides.toml
Connection failed, webserver is unreachable at URL:
http://localhost:8080/health
> Loaded config from: /fides/.fides/fides.toml
Connection failed, webserver is unreachable at URL:
http://localhost:8080/health
```